components/tours/hooks.ts

```diff
@@ -5,44 +5,72 @@ import {useCallback} from 'react';
 
 import {savePreferences} from 'mattermost-redux/actions/preferences';
 import {close as closeLhs, open as openLhs} from 'actions/views/lhs';
 import {setAddChannelDropdown} from 'actions/views/add_channel_dropdown';
 import {switchToChannels} from 'actions/views/onboarding_tasks';
+import {showRHSPlugin} from 'actions/views/rhs';
+import {setProductMenuSwitcherOpen} from 'actions/views/product_menu';
+
+import {useGetRHSPluggablesIds} from 'components/work_templates/hooks';
+
 import {getHistory} from 'utils/browser_history';
+import {suitePluginIds} from 'utils/constants';
+import {useProducts} from 'utils/products';
 import {GlobalState} from 'types/store';
 
 import {OnboardingTaskCategory, OnboardingTaskList, OnboardingTasksName} from '../onboarding_tasks';
 
-import {setProductMenuSwitcherOpen} from '../../actions/views/product_menu';
-
 import {
     CrtTutorialSteps,
     ExploreOtherToolsTourSteps,
     FINISHED,
     OnboardingTourSteps,
     TTNameMapToTourSteps,
     TutorialTourName,
+    WorkTemplateTourSteps,
 } from './constant';
 
 export const useGetTourSteps = (tourCategory: string) => {
     const isGuestUser = useSelector((state: GlobalState) => isCurrentUserGuestUser(state));
     const pluginsList = useSelector((state: GlobalState) => state.plugins.plugins);
-
+    const pluginProducts = useProducts();
+    const workTemplatesLinkedItems = useSelector((state: GlobalState) => getWorkTemplatesLinkedProducts(state));
     let tourSteps: Record<string, number> = TTNameMapToTourSteps[tourCategory];
 
+    let boardsProductEnabled = false;
+    let playbooksProductEnabled = false;
+    if (pluginProducts) {
+        boardsProductEnabled = pluginProducts.some((product) => (product.pluginId === suitePluginIds.focalboard) || (product.pluginId === suitePluginIds.boards));
+        playbooksProductEnabled = pluginProducts.some((product) => product.pluginId === suitePluginIds.playbooks);
+    }
+    const boardsPlugin = pluginsList.focalboard;
+    const playbooksPlugin = pluginsList.playbooks;
+
     if (tourCategory === TutorialTourName.EXPLORE_OTHER_TOOLS) {
-        const boards = pluginsList.focalboard;
-        const playbooks = pluginsList.playbooks;
         const steps: Record<string, number> = tourSteps as typeof ExploreOtherToolsTourSteps;
-        if (!playbooks) {
+        if (!playbooksPlugin && !playbooksProductEnabled) {
             delete steps.PLAYBOOKS_TOUR;
         }
 
-        if (!boards) {
+        if (!boardsPlugin && !boardsProductEnabled) {
+            delete steps.BOARDS_TOUR;
+        }
+        tourSteps = steps;
+    } else if (tourCategory === TutorialTourName.WORK_TEMPLATE_TUTORIAL) {
+        const steps: Record<string, number> = tourSteps as typeof WorkTemplateTourSteps;
+
+        if (workTemplatesLinkedItems.playbooks && workTemplatesLinkedItems.playbooks === 0) {
+            delete steps.PLAYBOOKS_TOUR;
+        }
```

Comment:
Could `workTemplatesLinkedItems.playbooks` ever be `null` and should we also delete the step in that case?

Reply:
Just to make sure I am following, if `workTemplatesLinkedItems.playbooks` is `null` that playbooks step was never added to begin with?

Reply:
on initial render workTemplatesLinkedItems is an empty object (https://github.com/mattermost/mattermost-webapp/pull/12219/files#diff-e75490b587f9fa072658f9d6b223817f8ad54af3f1c4716335b41d40e0b800fbR262), and it will keep that way until the executeTemplate is executed sucessfully, then the information about the items used by the template is summarized and filled in the workTemplatesLinkedItems object via a reducer placed in the global state (https://github.com/mattermost/mattermost-webapp/pull/12219/files#diff-14ff19a2b845055ffa501bef257b8f9ad99942f84dc252b6adef5b48f0e96b6aR270). So, to avoid initially deleting the step from the constant already loaded variable in memory, we first check that in that particular condition of the tourtip, for starting the information must be populated in the global state, specially before verifying the quantity and if the quantity is 0 then proceed to delete that step. Summarizing, in this particular scenario (the worktemplates tourtip), the initial global state is empty and we know and assume that on the subsequent render that value must have been populated before proceeding with the logic.